### ISSUE TITLE
InfoTypeConverter<double>::get(const curlpp::Easy & handle, CURLINFO info, double & value) returns garbage value

### ISSUE THREAD

**Issue by MtSimba** [jpbarrette/curlpp#109]
Hi! 

I have an issue when i want to use the CURL info methods that return a value in size double. 

When it is written like this i get return values of : **4.68087e+18**

```c++
template<>
void
InfoTypeConverter<double>::get(const curlpp::Easy & handle, 
				 CURLINFO info,
				 double & value)
{
  curl_off_t tmp;
  InfoGetter::get(handle, info, tmp);
  value = (double)tmp;
}
```
if i change the function to this :

```c++
template<>
void
InfoTypeConverter<double>::get(const curlpp::Easy & handle, 
				 CURLINFO info,
				 double & value)
{
  InfoGetter::get(handle, info, tmp);
}
```
It works fine. 

For example. I wanted to get CURLINFO_SIZE_UPLOAD. And with the new code i could get the correct value returned.

example : 

* upload completely sent off: 307 out of 307 bytes
< HTTP/1.1 204 No Content
< Content-Type: application/json
< Request-Id: 203e89dd-0e1a-11eb-8067-00224daa0347
< X-Influxdb-Build: OSS
< X-Influxdb-Version: 1.8.3
< X-Request-Id: 203e89dd-0e1a-11eb-8067-00224daa0347
< Date: Wed, 14 Oct 2020 12:38:13 GMT
< 
* Connection #0 to host 128.141.184.167 left intact

##### MY CODE #####
Uploaded: 307 bytes
##### MY CODE #####

So, my question is: Should this be fixed in the library in general or am i doing something wrong with the value? that is returned? Should i the return value from the get to something else or what should i do here? Or is it the library that might need a fix? 

The issue lies in ./src/curlpp/info.cpp

Thank you in advance

**Comment by sgallou** [comment]
Hi @MtSimba ,
I don't understand exactly your problem, because I don't understand your fix (tmp is not declared, and value is never set).
Can you explain to me, or add full example ?
Thanks,

**Comment by MtSimba** [comment]
I realised that I wrote tmp in the   InfoGetter::get function. What I wanted it to say was: 

```c++
template<>
void
InfoTypeConverter<double>::get(const curlpp::Easy & handle, 
				 CURLINFO info,
				 double & value)
{
  InfoGetter::get(handle, info, value);
}
```

When I used **curl_off_t** I did not get the expected result back. But when I just used the method I'm showing in the comment, then I do get the expected result back. 

So my question was : "Am I missing something in the implementation since **curl_off_t**  gives me a value I did not expect, while doing it the type conversion like for any other type in this library?" 

The way I produced the error was like this: 

```c++
try {
        curlpp::Cleanup cleaner;  // ensures RAII
        curlpp::Easy request;
        request.setOpt(new curlpp::options::Url(
        "http://localhost:8086/write?db="+std::string(hostname)+"&precision=ns"));
        request.setOpt(new curlpp::options::Verbose(true));
        request.setOpt(new curlpp::options::PostFields(query));
        request.setOpt(new curlpp::options::PostFieldSize(query.length()));
        request.perform();   

        double speed;
        curlpp::infos::SpeedUpload::get(request, speed);
        speed = speed/125000; //returns value in Mbit/s
        std::cout << "Upload speed: " << ul << " Mbit/s\n" << std::endl;     
    }
catch(curlpp::RuntimeError & e) {
    std::cout << e.what()) << std::end;
}
catch(curlpp::LogicError & e) {
    std::cout << e.what()) << std::end;
}
```

**Comment by sgallou** [comment]
Regarding Curl documentation, we maybe use deprecated entry point for SpeedUpload (CURLINFO_SPEED_UPLOAD), we should use CURLINFO_SPEED_UPLOAD_T instead.
Can you change it in `infos.hpp` and retry, please ?

Replace :
`typedef curlpp::Info<CURLINFO_SPEED_UPLOAD, double> SpeedUpload;`
by :
`typedef curlpp::Info<CURLINFO_SPEED_UPLOAD_T, double> SpeedUpload;`

and let me know.

Please note than you also can write your call in one line like that :
`double speed = curlpp::infos::SpeedUpload::get(request) / 12500;`

**Comment by MtSimba** [comment]
Thanks @sgallou ! That fixed the issue for me.

And thanks for the input on the example code. Will you patch this issue in the library? We're using curlpp in quite a lot of PC's at my office, and it would be great with a general patch for this specific issue.

**Comment by sgallou** [comment]
Probably, I'm just waiting the result of the test I ask to you.
If result is positive, it means that the problem exists for CURLINFO_SPEED_DOWNLOAD too.

**Comment by MtSimba** [comment]
The issue does also appear on CURLINFO_SPEED_DOWNLOAD, changing it to CURLINFO_SPEED_DOWNLOAD_T works also for download

**Comment by sgallou** [comment]
Fixed in develop branch by e87226626f07f7cf328ba2430ee547940fb47172
